core/src/test/java/org/apache/iceberg/TestMetricsConfig.java

```diff
@@ -28,18 +29,47 @@
 public class TestMetricsConfig {
 
   @Test
-  public void testNestedStructsRespectedInLimit() {
+  public void testNestedStruct() {
```

Comment:
better naming and numbering can help make it easier to understand. e.g. field ids can be numbered as 11, 12, ... for top level fields and 21, 22, ... for second level and so on.

field names can also be named a bit more informative: level1_s1, level1_s2, level1_a, level2_b, etc.